core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -97,61 +100,80 @@ private[spark] object JsonProtocol {
       case logStart: SparkListenerLogStart =>
         logStartToJson(logStart)
       case metricsUpdate: SparkListenerExecutorMetricsUpdate =>
-        executorMetricsUpdateToJson(metricsUpdate)
+        executorMetricsUpdateToJson(metricsUpdate, omitInternalAccums)
       case blockUpdated: SparkListenerBlockUpdated =>
         throw new MatchError(blockUpdated)  // TODO(ekl) implement this
       case _ => parse(mapper.writeValueAsString(event))
     }
   }
 
-  def stageSubmittedToJson(stageSubmitted: SparkListenerStageSubmitted): JValue = {
-    val stageInfo = stageInfoToJson(stageSubmitted.stageInfo)
+  def stageSubmittedToJson(
+    stageSubmitted: SparkListenerStageSubmitted,
+    omitInternalAccums: Boolean = false): JValue = {
+    val stageInfo = stageInfoToJson(stageSubmitted.stageInfo, omitInternalAccums)
     val properties = propertiesToJson(stageSubmitted.properties)
     ("Event" -> SPARK_LISTENER_EVENT_FORMATTED_CLASS_NAMES.stageSubmitted) ~
     ("Stage Info" -> stageInfo) ~
     ("Properties" -> properties)
   }
 
-  def stageCompletedToJson(stageCompleted: SparkListenerStageCompleted): JValue = {
+  def stageCompletedToJson(
+    stageCompleted: SparkListenerStageCompleted,
+    omitInternalAccums: Boolean = false): JValue = {
     val stageInfo = stageInfoToJson(stageCompleted.stageInfo)
```

Comment:
Yes, thank you for catching it. I think it got omitted while I was merging stuff. Will fix this.